lib/log.c

```diff
@@ -1151,3 +1151,35 @@ zlog_hexdump (const void *mem, unsigned int len) {
     }
     zlog_debug("\n%s", buf);
 }
+
+const char *
+zlog_sanitize (char *buf, size_t bufsz, const void *in, size_t inlen)
+{
+  const char *inbuf = in;
+  char *pos = buf, *end = buf + bufsz;
+  const char *iend = inbuf + inlen;
+
+  assert (bufsz >= inlen);
```

Comment:
In my opinion, this assert is not needed and should be removed. In my opinion it is a pefectly valid usecase of `zlog_sanitize` to dump the start of some binary content into a buffer in readable form.

The assert is superfluous, because because the first check in the for loop is `if (end - pos < 5)`  which is equivalent to `offset > bufsz - 5`. If this condition ocurrs, it will break from the loop without writing to the buffer. So the function will never overrun the output buffer, even without the assert.